lib/foreman_scap_client/client.rb

```diff
@@ -111,5 +100,43 @@ def foreman_proxy_uri
       foreman_proxy_port = config[:port]
       "https://#{foreman_proxy_fqdn}:#{foreman_proxy_port}"
     end
+
+    def generate_https_object(uri)
+      https = Net::HTTP.new(uri.host, uri.port)
+      https.use_ssl = true
+      https.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      https.ca_file = config[:ca_file]
+      begin
+        https.cert = OpenSSL::X509::Certificate.new File.read(config[:host_certificate])
+        https.key = OpenSSL::PKey::RSA.new File.read(config[:host_private_key])
+      rescue StandardError => e
+        puts 'Unable to load certs'
+        puts e.message
+        exit(3)
+      end
+      https
+    end
+
+    def ensure_scan_file
+      return if File.exist?(config[@policy_id][:content_path])
+      puts "File #{config[@policy_id][:content_path]} is missing. Downloading it from proxy"
+      begin
+        FileUtils.mkdir_p(File.dirname(config[@policy_id][:content_path]))
+        uri = URI.parse(download_uri(config[@policy_id][:download_path]))
+        puts "Download scap content xml from: #{uri}"
+        request = generate_https_object(uri).get(uri.path)
+        scap_content_xml = request.body
```

Comment:
we should check http status code here and save only if the status is OK, otherwise we save error pages as scap content file and we do not try to redownload content next time (I hit this many times during testing)